src/backend/common/helpers/matchstats_helper.py

```diff
@@ -72,184 +191,51 @@ def build_Minv_matrix(
                     team1_id = team_id_map[team1[3:]]
                     for team2 in alliance_teams:
                         M[team1_id, team_id_map[team2[3:]]] += 1
+
         return np.linalg.pinv(M)
 
     @classmethod
-    def build_s_matrix(
+    def __build_s_matrix(
         cls,
         matches: List[Match],
         team_id_map: TTeamIdMap,
-        stat_type: StatType,
-        init_stats: Optional[Dict[StatType, Dict[TeamId, int]]] = None,
-        init_stats_default: int = 0,
-        limit_matches: Optional[int] = None,
-    ) -> "np.ndarray[int]":
+        point_accessor: Callable[[Match, AllianceColor], float],
+    ) -> "np.ndarray[float]":
         n = len(team_id_map.keys())
         s = np.zeros((n, 1))
         for match in matches:
             if match.comp_level != CompLevel.QM:  # only consider quals matches
                 continue
 
-            treat_as_unplayed = (
-                limit_matches is not None and match.match_number > limit_matches
-            )
-
             for alliance_color in ALLIANCE_COLORS:
                 alliance_teams = [
                     team[3:] for team in match.alliances[alliance_color]["teams"]
                 ]
-                stat = cls._get_stat(
-                    stat_type,
-                    match,
-                    alliance_color,
-                    alliance_teams,
-                    init_stats,
-                    init_stats_default,
-                    treat_as_unplayed,
-                )
+
+                stat = point_accessor(match, alliance_color)
                 for team in alliance_teams:
-                    s[team_id_map[team]] += stat
+                    s[team_id_map[team]] += float(stat)
```

Comment:
nit: We shouldn't need to cast here, if the return type for this specifies it's a float. Why are we casting to a float?

Reply:
The reason was because some stats are returns as `int`s, but I'll change that behavior elsewhere to obey the typing better. (By making the default StatAccessor guarantee a float return)